File: typescript-api/src/data/services/update-todo.ts
import { Todo } from "../../domain/entities";
import { UpdateTodoRepository } from "../repositories";

export class UpdateTodoService {
  constructor(
    private readonly repo: UpdateTodoRepository
  ) {}

  async update(todo: Todo): Promise<void> {
    await this.repo.update(todo)
  }
}

File: rust-api/src/data/services/update_todo.rs
use std::sync::Arc;
use crate::data::{
    repositories::UpdateTodoRepository,
    dto::UpdateTodoDto
};

pub async fn update_todo_service(
    repo: Arc<impl UpdateTodoRepository>,
    new_todo: &UpdateTodoDto,
    todo_id: i32,
) -> Result<(), sqlx::Error> {
    repo.update(new_todo, todo_id).await
}
